### PR TITLE
Functions integration tests

### PR DESCRIPTION
**Pull request by dcermak**
Supersedes https://github.com/OSInside/kiwi/pull/1827

### CODE REVIEW

**Comment by schaefi** [comment]
@dcermak can you please review #1853 before making further force pushes here which produces conflicts ? It's hard to collaborare/incorporate changes that way ;) Thanks

**Comment by dcermak** [comment]
> @dcermak can you please review #1853 before making further force pushes here which produces conflicts ? It's hard to collaborare/incorporate changes that way ;) Thanks

I'd like to get the CI working first before throwing in more changes, otherwise that all requires manual verification.

File: doc/source/contributing/scripts_testing.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+This can be achieved by leveraging pytest's `fixture parametrization
+<https://docs.pytest.org/en/stable/parametrize.html>`__ as follows:
+
+.. code:: python
+
+    @pytest.mark.parametrize(
+        "shared_container",
+        (
+            "Tumbleweed",
+            "Leap-15.2",
```
Comment (schaefi) [inline doc/source/contributing/scripts_testing.rst:70]: Can we avoid using distro names that will be outdated and some point in the future ? I see two options:

1. Just stick with Tumbleweed
2. Use a variable for Leap-15.2 in ```doc/source/conf.py``` in the ```prolog_replacements``` and reference it here as {...}

Comment (dcermak) [reply doc/source/contributing/scripts_testing.rst]: This should be fixed now. I've reused the variable `exc_os_version` as there's not much to parametrize if you have only one OS :wink:

**Pull request merged**